### PR TITLE
annotate errors in apps/cronjob e2e tests

### PR DESCRIPTION
**Pull request by ajatprabha**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR annotates the errors in apps/cronjob e2e tests providing more context.

**Which issue(s) this PR fixes**:
WIP for #34059

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/apps/cronjob.go [hunk 1]
```diff
@@ -85,20 +85,20 @@ var _ = SIGDescribe("CronJob", func() {
...
 
 		By("Ensuring no jobs are scheduled")
 		err = waitForNoJobs(f.ClientSet, f.Namespace.Name, cronJob.Name, false)
-		Expect(err).To(HaveOccurred())
+		Expect(err).To(HaveOccurred(), "failed to ensure no scheduled jobs for CronJob %s in namespace: %s", cronJob.Name, f.Namespace.Name)
 
 		By("Ensuring no job exists by listing jobs explicitly")
 		jobs, err := f.ClientSet.BatchV1().Jobs(f.Namespace.Name).List(metav1.ListOptions{})
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to create JobList in namespace: %s", f.Namespace.Name)
```
Comment (oomichi) [inline test/e2e/apps/cronjob.go]: This is not for creating JobList, just for listing jobs as line 94.

File: test/e2e/apps/cronjob.go [hunk 2]
```diff
@@ -107,20 +107,20 @@ var _ = SIGDescribe("CronJob", func() {
...
 		By("Ensuring exactly one is scheduled")
 		cronJob, err = getCronJob(f.ClientSet, f.Namespace.Name, cronJob.Name)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to get CronJob %s in namespace: %s", cronJob.Name, f.Namespace.Name)
 		Expect(cronJob.Status.Active).Should(HaveLen(1))
 
 		By("Ensuring exactly one running job exists by listing jobs explicitly")
 		jobs, err := f.ClientSet.BatchV1().Jobs(f.Namespace.Name).List(metav1.ListOptions{})
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to create JobList in namespace: %s", f.Namespace.Name)
```
Comment (oomichi) [inline test/e2e/apps/cronjob.go]: ditto

File: test/e2e/apps/cronjob.go [hunk 3]
```diff
@@ -139,30 +139,30 @@ var _ = SIGDescribe("CronJob", func() {
...
```
Comment (oomichi) [inline test/e2e/apps/cronjob.go]: ditto

File: test/e2e/apps/cronjob.go [hunk 4]
```diff
@@ -234,37 +234,37 @@ var _ = SIGDescribe("CronJob", func() {
...
 
 		By("Ensuring a finished job exists")
 		err = waitForAnyFinishedJob(f.ClientSet, f.Namespace.Name)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to ensure a finished job in namespace: %s", f.Namespace.Name)
 
 		By("Ensuring a finished job exists by listing jobs explicitly")
 		jobs, err := f.ClientSet.BatchV1().Jobs(f.Namespace.Name).List(metav1.ListOptions{})
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to create JobList in namespace: %s", f.Namespace.Name)
```
Comment (oomichi) [inline test/e2e/apps/cronjob.go]: ditto

File: test/e2e/apps/cronjob.go [hunk 5]
```diff
@@ -234,37 +234,37 @@ var _ = SIGDescribe("CronJob", func() {
...
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to ensure Job is finished in namespace: %s", f.Namespace.Name)
 		err = waitForJobsPodToDisappear(f.ClientSet, f.Namespace.Name, finishedJobs[0])
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to ensure Pod disappearance in namespace: %s", f.Namespace.Name)
 
 		By("Ensuring there is 1 finished job by listing jobs explicitly")
 		jobs, err = f.ClientSet.BatchV1().Jobs(f.Namespace.Name).List(metav1.ListOptions{})
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to create JobList in namespace: %s", f.Namespace.Name)
```
Comment (oomichi) [inline test/e2e/apps/cronjob.go]: ditto

**Review by oomichi — COMMENTED** [review]
Most part seems good for me, just few comments

**Comment by oomichi** [comment]
Thanks for updating

/ok-to-test

**Comment by oomichi** [comment]
Oops, https://github.com/kubernetes/kubernetes/pull/71922 covers this change and that was subumitted before this.

**Comment by ajatprabha** [comment]
@oomichi I missed that too! 😅 I'll close this PR then and pick another section. Thanks!

**Pull request closed without merging**